satpy/composites/__init__.py

```diff
@@ -337,20 +337,28 @@ class CategoricalDataCompositor(CompositeBase):
         res = [[20, 40, 30], [50, 30, 10]]
     """
 
-    def __init__(self, name, lut=None, **kwargs):
+    def __init__(self, name, lut=None, dtype=None, fill_value=None, **kwargs):
         """Get look-up-table used to recategorize data.
 
         Args:
-            lut (list): a list of new categories. The lenght must be greater than the
-                        maximum value in the data array that should be recategorized.
+            lut (list):
+                A list of new categories. The lenght must be greater than the
+                maximum value in the data array that should be recategorized.
+            dtype (dtype, optional):
+                Set this to force the output dtype to be different from the input dtype.
+            fill_value (number):
+                Fill value to set.  This will be defined as an attribute to the data
+                returned when the compositor is called.
```

Comment:
Can we mention `_FillValue` here so there is no confusion what is being added?

Reply:
Can/should we provide a default `_FillValue`? This would save people from getting a warning when resampling if the integer data didn't have a _FillValue. We could use the input's _FillValue if it exists in `.attrs` or (similar to resampling) default to the maximum value for the dtype (integers only).

Reply:
I considered a default `_FillValue`, which could be taken from `netCDF4.default_fillvals`:

```
In [115]: netCDF4.default_fillvals
Out[115]:
{'S1': '\x00',
 'i1': -127,
 'u1': 255,
 'i2': -32767,
 'u2': 65535,
 'i4': -2147483647,
 'u4': 4294967295,
 'i8': -9223372036854775806,
 'u8': 18446744073709551614,
 'f4': 9.969209968386869e+36,
 'f8': 9.969209968386869e+36}
```

but then I decided against it, because maybe there are good reasons to not introduce a fill value in recategorised data if there isn't any in the input data.

Reply:
I also like the idea of a default `_FillValue`. In case this is not desired the user can always do `fill_value=None`? That could in that case be clearly documented. One could also print some log message about setting the `_FillValue` attribute in order inform the user.